documentation/correctness-previous.adoc

```diff
@@ -0,0 +1,130 @@
+= Model correctness
+
+We can identify a number of separate criteria for the correctness of a model.
+Failure to comply to one or more of these criteria doesn't imply the model as a whole is completely incorrect.
+Typically, it's quite possible to recover from such failures.
+The criteria can be visualized as follows.
+
+.The criteria for model correctness as a diagram
+image::../resources/correctness-diagrams/criteria.svg[align="center"]
+
+The discussion of these criteria below is divided into two layers:
+
+. JSON serialization of the model: the _serialization chunk_
+. The model (post-deserialization/in-memory)
+
+Each criterion has three aspects:
+
+. What are the conditions to comply to?
+. How is an incorrectness to be reported?
+. Can an incorrectness be recovered from (and if so: how)?
+
+
+== Serialization chunk
+
+=== Low-level syntactic
+
+The serialization is parsable as JSON.
+Note that “parsable” doesn't necessarily mean “syntax-correct”.
+E.g., trailing commas are not syntax-correct, but don't necessarily prevent the JSON from being parsable, nor deserializable.
+It's desirable to be able to recover from "`simple`"/"`obvious`" syntax errors in the JSON.
+
+Incorrectness is to be reported in terms of the location where parsing the JSON text failed, including an indication whether this was recoverable.
+The reporting is typically dependent on the precise JSON parser used.
+In particular, the JSON parser used might recover automatically, essentially ignoring syntax errors without reporting them in any way.
+This could be problematic, e.g. in case of duplicate keys in an object: the JSON parser is most likely to only reproduce the last value for the duplicate key, ignoring the previous ones.
+
+=== Structural syntactic
+
+The JSON conforms to https://github.com/LionWeb-io/specification/blob/main/serialization/serialization.adoc[the specification for the JSON serialization format].
+Note that JSON Schema isn't powerful enough to express more high-level, nor constraints of a referential nature on the serialization chunk.
+
+Incorrectness is to be reported in terms of e.g. a https://github.com/json-path/JsonPath[JsonPath] to the location(s) of failure(s).
+This is typically dependent on the precise JSON Schema validator framework/library/utility used.
+Recovering from an incorrectness is sometimes possible, e.g.:
+
+. If extra properties are present, they can simply be removed or ignored.
+. If properties are missing, they could just be added, provided they have a compliant default value.
+
+=== Hierarchical
+
+The JSON doesn't violate the following constraints.
+
+. IDs of serialized nodes are unique (within the serialization).
+Violation is to be reported in terms of precise locations in the JSON file of all the serialized nodes with equal ID.
```

Comment:
Here do we mean position as in a couple of points indicating line and column for each one or in terms of JSON path?
If we use position, do we start counting from Line 1 and Column 0?

Reply:
I wrote additional text on how to report locations.

And I would say: (1, 1).

Reply:
On this I think ANTLR would start from (1,0).

For example, If I have the text:
```
abc
def
```

`a` would be at Line 1, Column 0. Column 1 would be right after the first character, which seems to make sense to me. I think whatever solution we pick it would be confusing to some, but I would lean towards making the same choice as ANTLR